File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let longestSeq = 0;
        let currentSeq = 0
        let setOfNums = new Set(nums);
        for(let num of setOfNums){
            if(!setOfNums.has(num-1)){
                let currentNum = num;
                currentSeq = 0;
                while(setOfNums.has(currentNum)){
                    currentNum++;
                    currentSeq += 1;
                }
                longestSeq = Math.max(longestSeq, currentSeq);
            }
        }
        return longestSeq;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currentSum = nums[0];
        maxSum = nums[0]
        for i in range(1, len(nums)):
            if currentSum + nums[i] > nums[i]:
                currentSum = currentSum + nums[i]
            else: 
                currentSum = nums[i]

            maxSum = max(maxSum, currentSum)

        return maxSum;
        